admin-dev/themes/new-theme/js/components/grid/extension/async-submit-bulk-action-extension.js

```diff
@@ -0,0 +1,238 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import ConfirmModal from '../../modal';
+
+const {$} = window;
+
+/**
+ * Performs submit actions by partially sending data to the server.
+ */
+export default class AsyncSubmitBulkActionExtension {
+  /**
+   *
+   * @param {function|null} successCallback - if defined it overrides current success callback. Accepts response object
+   * as argument
+   * @param {function|null} errorCallback - if defined it overrides current error callback. Accepts response object
+   * as argument
+   * @param {function|null} progressCallback - if defined it tracks progress of executed ajax calls. First parameter
+   * is response object.Second parameter is currently sent ids and last parameter is the total ids array
+   * @return {{extend: (function(*=): void)}}
+   */
+  constructor({
+    successCallback = null,
+    errorCallback = null,
+    progressCallback = () => {},
+  } = {}) {
+    this.progressCallback = progressCallback;
+    this.successCallback = this.setSuccessCallback(successCallback);
+    this.errorCallback = this.setErrorCallback(errorCallback);
+
+    return {
+      extend: (grid) => this.extend(grid),
+    };
+  }
+
+  setSuccessCallback(callback) {
+    return this.isCallback(callback) ? callback
+      : (response) => {
+        if ('message' in response) {
+          window.showSuccessMessage(response.message);
+        }
+
+        window.location.reload();
+      };
+  }
+
+  setErrorCallback(callback) {
+    return this.isCallback(callback) ? callback
+      : (response) => {
+        if ('message' in response) {
+          window.showErrorMessage(response.message);
+        }
+
+        window.location.reload();
+      };
+  }
+
+  isCallback(callback) {
+    return callback && typeof callback === 'function';
+  }
+
+  /**
+   * Extend grid with bulk action submitting
+   *
+   * @param {Grid} grid
+   */
+  extend(grid) {
+    grid.getContainer().on('click', '.js-bulk-action-submit-async-btn', (event) => {
+      this.submit(event, grid);
+    });
+  }
+
+  /**
+   * Collects required data and submits the form.
+   * @param event
+   * @param grid
+   */
+  submit(event, grid) {
+    const $submitBtn = $(event.currentTarget);
+    const confirmMessage = $submitBtn.data('confirm-message');
```

Comment:
Please use camel cas here: `$submitBtn.data('confirmMessage')`

Reply:
Why? :thinking: 
![image](https://user-images.githubusercontent.com/1462701/73660655-e6d65180-4698-11ea-965b-ba8531a40664.png)
Both are valid.
But IMHO, using `confirm-message` is easier to find, because it's the same as the html `data-confirm-message`, unless it's `data-confirmMessage`

Reply:
I know both notations work, but since we use the `data()` function which is an interface on the dataset I think it's preferable to emphasize its use (I think dataset in vanilla code requried the use of camel case)
Or if you hate camel case so much you could also use `attr('data-confirm-message')` ^^ (even though I'm aware they are not strictly equivalent)
Anyway it's not that important which cas we use but at least if not consistent in the whole project (which is lame..) let's keep it consistent in the same file!
```js
const confirmMessage = $submitBtn.data('confirm-message');
const confirmTitle = $submitBtn.data('confirmTitle');
```

We should also make the same modif in the original extension by the way: https://github.com/PrestaShop/PrestaShop/blob/develop/admin-dev/themes/new-theme/js/components/grid/extension/submit-bulk-action-extension.js#L61

Reply:
You are right, we should follow to RFC convention
```
- the prefix data- is removed (including the dash);
- for any dash (U+002D) followed by an ASCII lowercase letter a to z, the dash is removed and the letter is transformed into its uppercase counterpart;
- other characters (including other dashes) are left unchanged.
```